src/character.cpp

```diff
@@ -1200,7 +1203,14 @@ int Character::best_nearby_lifting_assist() const
 int Character::best_nearby_lifting_assist( const tripoint &world_pos ) const
 {
     const quality_id LIFT( "LIFT" );
-    return std::max( { this->max_quality( LIFT ),
+    int mech_lift = 0;
+    if( has_effect( effect_riding ) && g->u.mounted_creature ) {
+        auto mons = g->u.mounted_creature.get();
+        if( mons->has_flag( MF_RIDEABLE_MECH ) ) {
+            mech_lift = mons->type->mech_str_bonus + 10;
```

Comment:
Maybe you should expose the mech properties of `mtype` via member functions in `monster`, so one does not have to access the monster type.